components/marketing_signup_modal/middleware.coffee

```diff
@@ -1,10 +1,17 @@
-url = require 'url'
+_ = require 'underscore'
 
 module.exports = (req, res, next) ->
   sd = res.locals.sd
+  slug = req.query['m-id']
+  modalData = _.findWhere(sd.MARKETING_SIGNUP_MODALS, { slug: slug })
+
+  if modalData
+    sd.MARKETING_SIGNUP_MODAL_IMG = modalData.image
```

Comment:
This is a clever way to inject data into `sd` to leverage the existing code, but be careful because `sd` is a global object. On the server that means it's shared across requests and can cause unexpected behaviors, or worse, leak user-specific data across requests if you're not careful.

You could address this by simply [modifying the request level `sd`](https://github.com/artsy/sharify#bootstrapping-request-level-data-to-the-client) via `res.locals.sd. MARKETING_SIGNUP_MODAL_IMG = ...` or refactor the code downstream so that you're passing the modal data through locals e.g. `res.locals.modal = modalData` and changing the template to use `modal.image` `modal.copy` etc. to be morel like the refactor we did in Force.

Either works for me, but the latter might be nicer as it's more consistent with what we do in Force.